File: apps/admin-web/src/app/shared/feed.models.ts
export interface FeedItem {
  id: string;
  publishedAtUtc: string;
  sourceId: string;
  sourceName: string;
  title: string;
  summary: string | null;
  canonicalUrl: string | null;
  status: string;
  duplicateCount: number;
  // Decision fields
  decisionType: string | null;
  decidedByRuleId: string | null;
  decisionReason: string | null;
  decidedAtUtc: string | null;
  scheduledAtUtc: string | null;
  trustLevelSnapshot: number | null;
}

export interface FeedItemDetail {
  id: string;
  publishedAtUtc: string;
  ingestedAtUtc: string;
  sourceId: string;
  sourceName: string;
  title: string;
  summary: string | null;
  bodyText: string;
  originalText: string | null;
  canonicalUrl: string | null;
  language: string | null;
  status: string;
  duplicateCount: number;
  // Decision fields
  decisionType: string | null;
  decidedByRuleId: string | null;
  decisionReason: string | null;
  decidedAtUtc: string | null;
  scheduledAtUtc: string | null;
  trustLevelSnapshot: number | null;
  media: MediaItem[];
}

export interface MediaItem {
  id: string;
  mediaType: string;
  url: string;
  thumbUrl: string | null;
}

export interface FeedResponse {
  items: FeedItem[];
  total: number;
  page: number;
  pageSize: number;
}

export interface FeedQueryParams {
  fromUtc?: string;
  toUtc?: string;
  sourceId?: string;
  keyword?: string;
  status?: string;
  decisionType?: string;
  page?: number;
  pageSize?: number;
}

export interface Source {
  id: string;
  name: string;
  type: string;
  url: string | null;
  description: string | null;
  group: string | null;
  trustLevel: number;
  priority: number;
  isActive: boolean;
  createdAtUtc: string;
  lastFetchedAtUtc: string | null;
}

export interface IngestionResult {
  ok: boolean;
  sourcesProcessed: number;
  itemsInserted: number;
  duplicates: number;
  errors: string[] | null;
  byDecisionTypeCounts: { [key: string]: number } | null;
}

File: apps/api/HaberPlatform.Api/Models/FeedModels.cs
namespace HaberPlatform.Api.Models;

public record FeedItemDto(
    Guid Id,
    DateTime PublishedAtUtc,
    Guid SourceId,
    string SourceName,
    string Title,
    string? Summary,
    string? CanonicalUrl,
    string Status,
    int DuplicateCount,
    // Decision fields
    string? DecisionType,
    Guid? DecidedByRuleId,
    string? DecisionReason,
    DateTime? DecidedAtUtc,
    DateTime? ScheduledAtUtc,
    int? TrustLevelSnapshot
);

public record FeedItemDetailDto(
    Guid Id,
    DateTime PublishedAtUtc,
    DateTime IngestedAtUtc,
    Guid SourceId,
    string SourceName,
    string Title,
    string? Summary,
    string BodyText,
    string? OriginalText,
    string? CanonicalUrl,
    string? Language,
    string Status,
    int DuplicateCount,
    // Decision fields
    string? DecisionType,
    Guid? DecidedByRuleId,
    string? DecisionReason,
    DateTime? DecidedAtUtc,
    DateTime? ScheduledAtUtc,
    int? TrustLevelSnapshot,
    List<MediaDto> Media
);

public record MediaDto(
    Guid Id,
    string MediaType,
    string Url,
    string? ThumbUrl
);

public record FeedResponse(
    List<FeedItemDto> Items,
    int Total,
    int Page,
    int PageSize
);

public record IngestionRunResponse(
    bool Ok,
    int SourcesProcessed,
    int ItemsInserted,
    int Duplicates,
    List<string>? Errors,
    Dictionary<string, int>? ByDecisionTypeCounts
);

// Legacy source DTOs - use SourceModels.cs for new code
// Kept for backward compatibility with any existing code

[Obsolete("Use SourceListItemDto or SourceDetailDto from SourceModels.cs")]
public record SourceDto(
    Guid Id,
    string Name,
    string Type,
    string? Url,
    string? Description,
    string? Group,
    int TrustLevel,
    int Priority,
    bool IsActive,
    DateTime CreatedAtUtc,
    DateTime? LastFetchedAtUtc
);

[Obsolete("Use UpsertSourceRequest from SourceModels.cs")]
public record CreateSourceRequest(
    string Name,
    string Type,
    string? Url,
    string? Description,
    string? Group,
    int TrustLevel = 1,
    int Priority = 0
);

[Obsolete("Use UpsertSourceRequest from SourceModels.cs")]
public record UpdateSourceRequest(
    string? Name,
    string? Url,
    string? Description,
    string? Group,
    int? TrustLevel,
    int? Priority,
    bool? IsActive
);
